src/cmap/handshake/handshake_document.ts

```diff
@@ -0,0 +1,18 @@
+import type { Document } from 'bson';
```

Comment:
Can we move this into another file? I would like to refrain from adding type only files as it will generate an empty javascript files. I suggest removing `default_` from the base handshake decorator and we can have these types exported from that file.